openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -110,7 +113,16 @@ def load_transformed_data(
         """
         start_time = time.perf_counter()
         (batch_start, batch_end) = batch
-        pg = PostgresHook(postgres_conn_id)
+        # Considered using PostgresHook.get_execution_timeout(task), but not sure how
+        # that will work for mapped tasks. Is it total across all mappings or for each?
+        # This we know is for each iteration.
+        # In testing this locally, the longest full iteration took 39 minutes,
+        # median was 18 minutes. We should probably adjust the timeouts with
+        # more info from production runs.
+        pg = PostgresHook(
+            postgres_conn_id=POSTGRES_CONN_ID,
+            default_statement_timeout=timedelta(hours=1).total_seconds(),
+        )
```

Comment:
Yeah, it's just that the operator kwargs are stored a little differently in the [MappedOperator class](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/mappedoperator/index.html). But I added a branch to handle them separately.

[This](https://github.com/apache/airflow/blob/eb46eeb33d58436aa5860f2f0031fad3dea3ce3b/airflow/models/taskinstance.py#L463) is what makes me think that tasks are safer than task_instances for timeout purposes.

Reply:
`sql.clean_intermediate_table_data` and `sql.upsert_records_to_db_table` are already using the timeout from the task, right? Testing locally, it looks like it is working in those cases.

But doesn't this mean that each of the individual SQL commands that are part of this task all have a separate 1-hr timeout, that could theoretically end up getting hit sequentially? Is that intended/something we have to live with?

Reply:
Hmm, that's a good point. I think relying on the default task timeout here is OK, even if it means _cumulatively_ that the task may go over its official 1 hour limit. In the worst case, the task takes double the applied time out (2 hours) if we assume that a 1h timeout query is executed in the last minute of the task's total 1h timeout. Given that this is a significant improvement over no timeout at all and the query hanging indefinitely (or at least until it resolves), I'm fine leaving it as-is! Especially since it'd take some effort to parse out reasonable sub-timeouts for each command, which I'm not sure is worth it IMHO.

Reply:
Yeah, my assumption was that most of the time we could live with this for a first go. If there are places where we think a shorter timeout is in order, the task level timeout is just the default for the hook and we (but not me, honestly!) could always go back and use different `statement_timeout`s in the `run` commands within a given function.